### PR TITLE
[Report V2] Remove report v2 flags

### PR DESCRIPTION
**Pull request by tfrcarvalho**
# Description

* Removes `report_v2` feature flag
* Removes unreachable code in show action
* Rename v2 functions to drop v2 when the equivalent old function was dropped

# Tests

* Open report page and test all interactions
* WIP: test report precache
* WIP: test report csv download

### CODE REVIEW

File: app/assets/src/api/index.js [hunk 1]
```diff
@@ -224,7 +224,7 @@ const getSampleStats = ({ domain, filters, projectId, search }) =>
     },
   });
 
-const getSample = ({ sampleId }) => get(`/samples/${sampleId}/show_v2.json`);
+const getSample = ({ sampleId }) => get(`/samples/${sampleId}.json`);
```
Comment (j-x-han) [inline app/assets/src/api/index.js]: Do you also want to change the `getSampleReportData` and `getSampleReportInfo` methods earlier on in the file?

File: app/assets/src/components/views/SampleViewV2/SampleViewHeader.jsx [hunk 2]
```diff
@@ -105,7 +105,7 @@ export default function SampleViewHeader({
             label: sample.name,
             id: sample.id,
             onClick: () => {
-              openUrl(`/samples/${sample.id}/show_v2`, event);
+              openUrl(`/samples/${sample.id}/show`, event);
```
Comment (j-x-han) [inline app/assets/src/components/views/SampleViewV2/SampleViewHeader.jsx]: This causes an error when selecting another sample in the project from the dropdown, I think the route is just ``/samples/${sample.id}``

File: app/assets/src/api/index.js (hunk 1 above)
Comment (tfrcarvalho) [reply app/assets/src/api/index.js]: You mean deleting the `getSampleReportInfo`? Yes.
What is your suggestion regarding `getSampleReportData` ?

Comment (tfrcarvalho) [reply app/assets/src/api/index.js]: But I will probably delete on a future PR. At the same time I delete report_info on samples controller.

File: app/assets/src/components/views/SampleViewV2/SampleViewHeader.jsx (hunk 2 above)
Comment (tfrcarvalho) [reply app/assets/src/components/views/SampleViewV2/SampleViewHeader.jsx]: Done!

**Pull request merged**